PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h

```diff
@@ -2510,24 +2251,45 @@ void SteerRec(CollisionRec const& collision, TracksRec const& tracks)
   }
 
   // *) Do all thingies before starting to process data from this collision (e.g. count number of events, fetch the run number, etc.):
-  Preprocess(collision); // see enum eRecSim for possible values of 'rsb'
-
-  // *) Fill event histograms for reconstructed data before event cuts:
-  FillEventHistograms<eRec>(collision, tracks, eBefore);
+  Preprocess(collision);
+
+  // *) Fill event histograms before event cuts:
+  if (gProcessRec) {
+    FillEventHistograms<eRec>(collision, tracks, eBefore);
+  } else if (gProcessRecSim) {
+    FillEventHistograms<eRec>(collision, tracks, eBefore);
+    FillEventHistograms<eSim>(collision, tracks, eBefore);
+  } else if (gProcessSim) {
+    FillEventHistograms<eSim>(collision, tracks, eBefore);
+  }
```

Comment:
If you adapt your `FillEventHistograms` function for the `eRecSim` value you will need only a single function call in here. As was indicated before in the previous PR, with the full template mechanism you don't need the global variables which need to be set via configuration

Reply:
I need these global variables also for other reasons at the moment in the code. 

I can remove them eventually, but that will trigger a lot of other modifications in the code.